src/core/net/netif.hpp

```diff
@@ -163,6 +163,16 @@ class ExternalNetifMulticastAddress : public NetifMulticastAddress
     friend class LinkedList<ExternalNetifMulticastAddress>;
 
 public:
+    /**
+     * This enumeration defines external Netif multicast address filter.
+     *
+     */
+    enum Filter : uint8_t
+    {
+        kFilterAny,                  ///< Accept any external Netif multicast address.
+        kFilterLargerThanRealmLocal, ///< Accept external Netif multicast addresses with scope larger than Realm Local.
+    };
```

Comment:
The idea of adding this sounds good, but I suggest waiting for https://github.com/openthread/openthread/pull/5269 to be merged and then using the common definition in `Ip6::Address` for filter type.

Reply:
Using the same filter type sounds good to me.

Reply:
rebased on #5269  to use the same type filter